src/analytics.js

```diff
@@ -0,0 +1,204 @@
+  if (CI_SERVER_URL) return 'gitlab';
+  if (BITBUCKET_WORKSPACE) return 'bitbucket';
+  if (TF_BUILD) return 'azure';
+  if (CI) return 'unknown';
+  return '';
+};
+
+const isCI = () => {
+  const ci = guessCI();
+  return ci.length > 0;
+};
+
+const groupId = async () => {
+  if (!isCI()) return '';
+
+  const ci = guessCI();
+  let rawId = 'CI';
+  if (ci === 'github') {
+    rawId = `${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY_OWNER}`;
+  } else if (ci === 'gitlab') {
+    rawId = `${CI_SERVER_URL}/${CI_PROJECT_ROOT_NAMESPACE}`;
+  } else if (ci === 'bitbucket') {
+    rawId = BITBUCKET_WORKSPACE;
+  }
+
+  return await deterministic(rawId);
+};
+
+const userId = async ({ cml } = {}) => {
+  try {
+    if (isCI()) {
+      let rawId;
+      const ci = guessCI();
+      if (ci === 'github') {
+        const { name, login, id } = await cml
+          .getDriver()
+          .user({ name: GITHUB_ACTOR });
+        rawId = `${name || ''} ${login} ${id}`;
+      } else if (ci === 'gitlab') {
+        rawId = `${GITLAB_USER_NAME} ${GITLAB_USER_LOGIN} ${GITLAB_USER_ID}`;
+      } else if (ci === 'bitbucket') {
+        rawId = BITBUCKET_STEP_TRIGGERER_UUID;
+      } else {
+        rawId = await exec(`git log -1 --pretty=format:'%ae'`);
+      }
+
+      return await deterministic(rawId);
+    }
+
+    let id = uuidv4();
+    const oldPath = userConfigDir('dvc/user_id', 'iterative');
+    const newPath = userConfigDir('iterative/telemetry');
+    const readId = async ({ fpath }) => {
+      const content = (await fs.readFile(fpath)).toString('utf-8');
+      try {
+        const { user_id: id } = JSON.parse(content);
+        return id;
+      } catch (err) {
+        return content;
+      }
+    };
+
+    const writeId = async ({ fpath, id }) => {
+      await fs.mkdir(path.dirname(fpath), { recursive: true });
+      await fs.writeFile(fpath, JSON.stringify({ user_id: id }));
+    };
+
+    if (await fileExists(newPath)) {
+      id = await readId({ fpath: newPath });
+    } else {
+      if (await fileExists(oldPath)) id = await readId({ fpath: oldPath });
+
+      await writeId({ fpath: newPath, id });
+    }
+
+    if (!(await fileExists(oldPath)) && id !== ID_DO_NOT_TRACK)
+      await writeId({ fpath: oldPath, id });
+
+    return id;
+  } catch (err) {
+    winston.debug(`userId failure: ${err.message}`);
+  }
+};
+
+const OS = () => {
+  if (process.platform === 'win32') return 'windows';
+  if (process.platform === 'darwin') return 'macos';
+
+  return process.platform;
+};
+
+const jitsuEventPayload = async ({
+  action = '',
+  error = '',
+  extra = {},
+  cml
+} = {}) => {
+  try {
+    const { cloud: backend = '', ...extraRest } = extra;
+    extraRest.ci = guessCI();
+
+    return {
+      user_id: await userId({ cml }),
+      group_id: await groupId(),
+      action,
+      interface: 'cli',
+      tool_name: 'cml',
+      tool_version: VERSION,
+      tool_source: '',
+      os_name: OS(),
+      os_version: os.release(),
```

Comment:
different from Python's `distro.version()`?

Reply:
Definitely different. While some docs states that Im having the version as X.X.X current docs states that we are having the same as uname... For my mac

```
Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64
```

Reply:
It the same. In macos prints `19.6.0`

Reply:
clearly not the same:


```bash
$ node
> os.release()
'5.15.0-43-generic'
$ python
>>> import distro
>>> distro.version()
'22.04'
```

Reply:
@casperdcl 

<img width="512" alt="image" src="https://user-images.githubusercontent.com/414967/183637996-7f9c4112-f168-497a-8d3f-c02048f17dfd.png">

Reply:
It might be some discrepancies but it should be the same

Reply:
Well I had differences on a vanilla Ubuntu machine (which is what most CML jobs will be run on) so we need to fix this. Can you test on an ubuntu machine?

Reply:
Right now with [a7944f8](https://github.com/iterative/cml/pull/1092/commits/a7944f81505683753c08226ce72bfab049d57792)  on Macos 

```sh
os_name: 'macos',
os_version: '19.6.0',
```

on Ubuntu 
```sh
 os_name: 'linux',
 os_version: '20.04',
```

Reply:
For `os_name: 'macos'` the `os_version` is the Darwin Kernel version, not the operating system version.

Reply:
@0x2b3bfa0 where ? here or in the python version?

Reply:
In the comment you posted: https://github.com/iterative/cml/pull/1092#discussion_r942312221

Reply:
consistency fine by me too :)